plugin/trino-ranger/src/main/java/io/trino/plugin/ranger/RangerSystemAccessControl.java

```diff
@@ -0,0 +1,1391 @@
+    {
+        return new RangerTrinoResource(catalogName, schemaName, tableName, column);
+    }
+
+    private static List<RangerTrinoResource> createResource(CatalogSchemaTableName table, Set<String> columns)
+    {
+        List<RangerTrinoResource> colRequests = new ArrayList<>();
+
+        if (!columns.isEmpty()) {
+            for (String column : columns) {
+                RangerTrinoResource rangerTrinoResource = createResource(table.getCatalogName(), table.getSchemaTableName().getSchemaName(), table.getSchemaTableName().getTableName(), column);
+
+                colRequests.add(rangerTrinoResource);
+            }
+        }
+        else {
+            colRequests.add(createResource(table.getCatalogName(), table.getSchemaTableName().getSchemaName(), table.getSchemaTableName().getTableName(), null));
+        }
+
+        return colRequests;
+    }
+
+    private static RangerTrinoResource createResource(String catalogName, SchemaFunctionName functionName)
+    {
+        RangerTrinoResource res = new RangerTrinoResource();
+
+        res.setValue(RangerTrinoResource.KEY_CATALOG, catalogName);
+        res.setValue(RangerTrinoResource.KEY_SCHEMA, functionName.getSchemaName());
+        res.setValue(RangerTrinoResource.KEY_SCHEMA_FUNCTION, functionName.getFunctionName());
+
+        return res;
+    }
+
+    private static RangerTrinoResource createResource(EntityKindAndName entity)
+    {
+        RangerTrinoResource ret = new RangerTrinoResource();
+
+        switch (entity.entityKind().toUpperCase(ENGLISH)) {
+            case "SCHEMA":
+                ret.setValue(RangerTrinoResource.KEY_CATALOG, entity.name().getFirst());
+                ret.setValue(RangerTrinoResource.KEY_SCHEMA, entity.name().get(1));
+                break;
+
+            case "TABLE":
+            case "VIEW":
+            case "MATERIALIZED VIEW":
+                ret.setValue(RangerTrinoResource.KEY_CATALOG, entity.name().getFirst());
+                ret.setValue(RangerTrinoResource.KEY_SCHEMA, entity.name().get(1));
+                ret.setValue(RangerTrinoResource.KEY_TABLE, entity.name().get(2));
+                break;
+        }
+
+        return ret;
+    }
+
+    private static RangerTrinoResource createRoleResource(String roleName)
+    {
+        RangerTrinoResource res = new RangerTrinoResource();
+
+        res.setValue(RangerTrinoResource.KEY_ROLE, roleName);
+
+        return res;
+    }
+
+    private static Set<RangerTrinoResource> createRoleResources(Set<String> roleNames)
+    {
+        Set<RangerTrinoResource> ret = new HashSet<>(roleNames.size());
+
+        for (String rolName : roleNames) {
+            ret.add(createRoleResource(rolName));
+        }
+
+        return ret;
+    }
+
+    private static RangerTrinoResource createResource(QueryId queryId)
+    {
+        RangerTrinoResource res = new RangerTrinoResource();
+
+        res.setValue(RangerTrinoResource.KEY_QUERY_ID, queryId != null ? queryId.getId() : "*");
```

Comment:
Use `Optional` for nullable parameters

Reply:
`queryId` parameter could be null in the override of the deprecated method `checkCanExecuteQuery(Identity identy)` is called. Removed this override so that the default implementation in `SystemAccessControl.checkCanExecuteQuery(Identity identy)` will be executed - which throws `AccessDeniedException`,